plugins/package-managers/spm/src/main/kotlin/SpmModel.kt

```diff
@@ -99,31 +99,27 @@ data class AppDependency(
             }
     }
 
-    val vcs: VcsInfo
-        get() {
-            val vcsInfoFromUrl = VcsHost.parseUrl(repositoryUrl)
-
-            if (vcsInfoFromUrl.revision.isBlank() && state != null) {
-                when {
-                    !state.revision.isNullOrBlank() -> return vcsInfoFromUrl.copy(revision = state.revision)
-                    !state.version.isNullOrBlank() -> return vcsInfoFromUrl.copy(revision = state.version)
-                }
+    fun toPackage(): Package {
```

Comment:
Actually, this could be a regular / lazy property now, to avoid re-evaluation on each function call as the input properties (like `repositoryUrl`) never change.

Reply:
Why isn't the same done for `LibraryDependency`?

Reply:
For `LibraryDependency` the `id` property is used from elsewhere.
So, doing the exact same thing is not possible. 

The library dependency code of course could be changed as well. 
I didn't want to increase the commit count any further as I think the PR is already large enough.
And my primary concern is the lockfile bits.

Reply:
> For `LibraryDependency` the `id` property is used from elsewhere.

Yes, but that other place could have simply accessed the `id` via the package. 

> I didn't want to increase the commit count any further as I think the PR is already large enough.

As it follows the same idea, it would have been fine with me to do it in the same commit.

Anyway, I'll probably make a proposal once you're done with your refactoring.

Reply:
Thanks!

Reply:
> Anyway, I'll probably make a proposal once you're done with your refactoring.

So, @fviernau please let me know once you're done.

Reply:
Will do.